server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -105,17 +104,16 @@ package object server {
     def apply[F[_], Err, T](
         authUser: Kleisli[F, Request[F], Either[Err, T]],
```

Comment:
@rossabaker Since this may change source/binary compatibility, perhaps we can reduce this to a `Request[T] => F[Either[Err, T]]`

Reply:
I'm kind of surprised we didn't make it `Kleisli[EitherT[F, Err, ?], Request[F], T]`, which I'm sure goes in the opposite direction that you prefer.  I also haven't used our auth and don't have a great feel for the ergonomics.

Reply:
I suppose not, mostly because it is shorter and I do find it easier to read: 
 
```Scala
Request[F] => F[Either[Err, T]]

Kleisli[F, Request[F], Either[Err, T]]

Kleisli[EitherT[F, Err, ?], Request[F], T]
```

Anyway, that is a story for another time.